### DISCUSSION TITLE
Property 'upload.tool.' is undefined

### DISCUSSION THREAD

**Discussion by sdwood68** [espressif/arduino-esp32#8110] [General]
So, this keeps coming up or "Property 'upload.tool.serial' is undefined" evry time I update the Arduino IDE or the board support package.
I'm now using Arduino IDE 2.1 and arduino-esp32 2.0.8.

I've bbe able to fix this by hollowing the directions here https://www.bing.com/ck/a?!&&p=23bd195575ddcde8JmltdHM9MTY4MjEyMTYwMCZpZ3VpZD0wZTM0OWJjMy1mNGI4LTZjZGYtMGQ4NS04OTA4ZjUzNzZkNjYmaW5zaWQ9NTE5NQ&ptn=3&hsh=3&fclid=0e349bc3-f4b8-6cdf-0d85-8908f5376d66&psq=Property+%22upload.tool.%22+is+undefined&u=a1aHR0cHM6Ly9mb3J1bS5hcmR1aW5vLmNjL3QveGlhby1ib2FyZHMtcHJvcGVydHktdXBsb2FkLXRvb2wtc2VyaWFsLWlzLXVuZGVmaW5lZC8xMDM1MDEw&ntb=1

But why is'nt this being addressed in the board support files here?

**Comment by mrengineer7777** [comment]
@sdwood68 The boards.txt file should be modified with those fixes.  Contact your board manufacturer and ask them to submit a PR fixing their board entry.

**Comment by sdwood68** [comment]
I don't understand, the board.txt file from this repository is missing
".upload.tool.serial=esptool_py" on all the boards listed.  Wouldn't this
repository be responsible?

Stuart
